src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -150,16 +157,31 @@ void AddCoverageExclusions(string coveragePaths)
                 AddSetting(config.LocalSettings, SonarExclusions, localExclusions);
             }
         }
-    }
 
-    private static void AddSetting(AnalysisProperties properties, string id, string value)
-    {
-        var property = new Property(id, value);
+        private static string[] CoveragePaths(ProcessedArgs localSettings, IDictionary<string, string> serverProperties)
+        {
+            var localProperties = localSettings.AllProperties().ToList();
+            var coveragePaths = new List<string>
+            {
+                CoveragePaths(localProperties, serverProperties, VsCoverageReportsPaths),
+                CoveragePaths(localProperties, serverProperties, DotCoverReportsPaths),
+                CoveragePaths(localProperties, serverProperties, OpenCoverReportsPaths),
+            };
 
-        // Ensure it isn't possible to write sensitive data to the config file
-        if (!property.ContainsSensitiveData())
+            return coveragePaths.Where(x => x is not null).ToArray();
+        }
+
+        private static string CoveragePaths(List<Property> localProperties, IDictionary<string, string> serverProperties, string propertyName)
```

Comment:
This is the main difference.
We take into account local and server settings on an individual property level.
This will help when we implement the directory exclusion for `dotCover`.